kotlin/internal/jvm/compile.bzl

```diff
@@ -760,9 +760,9 @@ def _run_kt_java_builder_actions(
         java_infos.append(kt_java_info)
 
     # Build Java
-    # If there is Java source or KAPT generated Java source compile that Java and fold it into
+    # If there is Java source or KAPT/KSP generated Java source compile that Java and fold it into
     # the final ABI jar. Otherwise just use the KT ABI jar as final ABI jar.
-    if srcs.java or generated_kapt_src_jars or srcs.src_jars:
+    if srcs.java or generated_kapt_src_jars or srcs.src_jars or generated_ksp_src_jars:
```

Comment:
Doesn't this essentially mean that all targets that execute KSP will be forced to run a Java compilation action even if the underlying KSP plugin only produces Kotlin or class file outputs?

Reply:
Great point, that's correct. 

I'm thinking how we could avoid that and I'm not really sure how we could get the information if the plugin is generating java code or not, without getting this provided as one of `kt_ksp_plugin` parameters. 
Could be something like
```python
kt_ksp_plugin = rule(
    attrs = {
        "generates_java": attr.bool(
                doc = """Runs Java compilation action for plugin generating Java output"""",
                default = False,
        ),
       ...
    }
    ...
}    
``` 
What do you think about this approach? Or maybe there is some detail information that I'm missing, that could be used to determine if we could enforce/avoid running Java compilation action for certain KSP plugins? 👀

Reply:
@Bencodes updated PR with the approach mentioned above